packages/flutter_tools/lib/src/resident_runner.dart

```diff
@@ -483,8 +485,10 @@ abstract class ResidentRunner {
   }
 
   Future<void> refreshViews() async {
+    final List<Future<void>> futures = <Future<void>>[];
     for (FlutterDevice device in flutterDevices)
-      await device.refreshViews();
+      futures.add(device.refreshViews());
+    await Future.wait(futures);
```

Comment:
This is the change that breaks named_isolates_test.
From what I can tell flutter engine is getting stuck processing two concurrent listViews requests https://github.com/flutter/engine/blob/master/runtime/service_protocol.cc#L237.

cc @jason-simmons @chinmaygarde

Reply:
To reproduce the failure pick up this new implementation of refreshViews() method and run named_isolates_test locally from `$FH/flutter/dev/devicelab`: `$FH/flutter/bin/cache/dart-sdk/bin/dart bin/run.dart -t named_isolates_test`

Reply:
I am fixing refreshViews so it is app-wide, rather than isolate-specific message, that should fix deadlock in the engine((in https://github.com/flutter/flutter/commit/89665ac4527fd8e9cadc1df28c86c9b0c8659a58). @chinmaygarde is making engine more robust to these kind of scenarios.

Reply:
awesome.